homeassistant/components/ourgroceries/__init__.py

```diff
@@ -0,0 +1,48 @@
+"""The OurGroceries integration."""
+from __future__ import annotations
+
+from asyncio import TimeoutError as AsyncIOTimeoutError
+
+from aiohttp import ClientError
+from ourgroceries import OurGroceries
+from ourgroceries.exceptions import InvalidLoginException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import OurGroceriesDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.TODO]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up OurGroceries from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    data = entry.data
+    og = OurGroceries(data[CONF_USERNAME], data[CONF_PASSWORD])
+    lists = []
+    try:
+        await og.login()
+        lists = (await og.get_my_lists())["shoppingLists"]
+    except (AsyncIOTimeoutError, ClientError, InvalidLoginException) as error:
```

Comment:
I would say in the case of InvalidLoginException, raise a ConfigEntryError or return False. Otherwise it will keep retrying to log in with invalid credentials